customers/schema/mutations.py

```diff
@@ -1,110 +1,167 @@
 import graphene
 from django.core.exceptions import ValidationError
-from django.db import transaction
+from django.db import IntegrityError, transaction
 from graphene_file_upload.scalars import Upload
 
 from berth_reservations.exceptions import VenepaikkaGraphQLError
+from resources.models import BoatType
 from users.decorators import (
     add_permission_required,
     change_permission_required,
     delete_permission_required,
 )
-from utils.relay import get_node_from_global_id
+from utils.relay import from_global_id, get_node_from_global_id
 from utils.schema import update_object
 
-from ..models import BoatCertificate
-from .types import BoatCertificateNode, BoatCertificateTypeEnum, BoatNode
+from ..models import Boat, BoatCertificate
+from .types import BoatCertificateTypeEnum, BoatNode, ProfileNode
 
 
-class BoatCertificateInput:
+def add_boat_certificates(certificates, instance):
+    try:
+        for cert_input in certificates:
+            BoatCertificate.objects.create(**cert_input, boat=instance)
+    except IntegrityError as e:
+        raise VenepaikkaGraphQLError(e)
+
+
+def remove_boat_certificates(certificates, instance):
+    try:
+        for cert_id in certificates:
+            cert_id = from_global_id(cert_id)
+            BoatCertificate.objects.get(pk=cert_id, boat=instance).delete()
+    except BoatCertificate.DoesNotExist as e:
+        raise VenepaikkaGraphQLError(e)
+
+
+class BoatCertificateInput(graphene.InputObjectType):
     file = Upload()
     certificate_type = BoatCertificateTypeEnum()
     valid_until = graphene.Date()
     checked_at = graphene.Date()
     checked_by = graphene.String()
 
 
-class CreateBoatCertificateMutation(graphene.ClientIDMutation):
-    class Input(BoatCertificateInput):
-        boat_id = graphene.ID(required=True)
-        certificate_type = BoatCertificateTypeEnum(required=True)
+class BoatInput:
```

Comment:
not that it has much effect, but should this also extend `graphene.InputObjectType`?

Reply:
Nope. The `mutation.Input` automatically extends `InputObjectType`, so `BoatInput` doesn't have to.
```python
class CreateBoatMutation(graphene.ClientIDMutation):
    class Input(BoatInput):
        pass
```
If you extend `InputObjectType` both on `BoatInput` and then `mutation.Input`, you get a resolution problem because python doesn't know the hierarchy of the extends.

```
TypeError: Cannot create a consistent method resolution
order (MRO) for bases InputObjectType, Input
```

`BoatCertificateInput` requires extending the `InputObjectType` because it's used for input but it doesn't directly extend the input superclass.

Reply:
ahhh, gotcha! 👍